### PR TITLE
updated assertion_filter_cond by adding gsub() argument to line 520 s…

### PR DESCRIPTION
**Pull request by ProfessorP-beep**
…o that the error message returned for arg_name returns inputs wrapped in functionsby themselves without the function. So instead of expr(filter_add) in the error message its filter_add

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/dev/CONTRIBUTING.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by ProfessorP-beep** [comment]
@bms63 Hey Ben, I'm getting some errors about dependencies when doing the checks. It looks like its on Github's end but want to make sure I'm not missing anything.

File: R/assertions.R [hunk 1]
```diff
@@ -517,6 +517,8 @@ assert_filter_cond <- function(arg,
     return(invisible(arg))
   }
 
+  arg_name <- gsub("^.*\\((.*)\\)$", "\\1", arg_name)
```
Comment (bms63) [inline R/assertions.R:520]: @bundfussr and @ddsjoberg  is there another way to do this using something from rlang? 
image

**Comment by bms63** [comment]
@cicdguy we have a bunch of actions failing cause of staged dependencies.  Did we need to update something?

**Comment by bms63** [comment]
> Can we try removing `admiral.test` from https://github.com/pharmaverse/admiralvaccine/blob/main/staged_dependencies.yaml?

this is admiraldev?

File: R/assertions.R (hunk 1 above)
Comment (ddsjoberg) [reply R/assertions.R:520]: rlang has functions for working with calls, but not strings. If you have a call, you can use `rlang::call_args()` to get the arguments of the call.

``` r
my_call <- rlang::expr(rlang::enquo(data))
my_call
#> rlang::enquo(data)
```

``` r
rlang::call_args(my_call)
#> [[1]]
#> data
```

<sup>Created on 2024-05-17 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by ProfessorP-beep** [comment]
@cicdguy Yea sure, go ahead.

File: R/assertions.R (hunk 1 above)
Comment (ProfessorP-beep) [reply R/assertions.R:520]: I may be able to put it further in the function after call_args if we can't replace it in the arguments, but rlang::expr_text() returned 'my_expression' in place of 'filter_add' when generating the error. I thought that could work, but don't know what everyone else thinks.

**Comment by ProfessorP-beep** [comment]
@ccidguy @bms63 Just checking in if its safe to try another push? Also, if its possible to use rlang::expr_text() in place of rlang:call_args() which would give an error back for the variable assigned to filter_add, e.g. "Argument 'my_expression' should be a filter condition". I think I can make some changes that aren't gsub that could return the same thing. It just wouldn't use 'filter_add'

**Comment by bms63** [comment]
@bundfussr can you also weigh in here please

**Comment by bundfussr** [comment]
> @bundfussr can you also weigh in here please

From my side it is fine to use `gsub()`. However, I wonder if the condition for modifying `arg_name` should be more restrictive, e.g, `arg_name <- gsub("^enexpr\\((\\w+)\\)$", "\\1", arg_name)`.

We should do the same update for the other assertions which are usually called with `arg = enexpr(arg)`, i.e., `assert_symbol()` and `assert_expr()`.
And the new behavior needs to be documented.

**Comment by ddsjoberg** [comment]
Hi All! I was thinking about this PR last night, and wanted to share two thoughts:
- If we go the route of using `gsub()`, that should go into the default `arg_name` argument value, rather than into the body of the function. Then if a user ever wants to override the `arg_name` default, the passed value will not be subject to additional parsing.
- Speaking of the `arg_name` argument: My preference here is now to just do nothing. If we are calling this function AND changing the argument as it's passed to the function (e.g. wrapping the true argument name in `enexpr()`), we can just specify the `arg_name` value. Then this function remains in alignment with every other checking function. For example, `assert_filter_cond(exprs(filter_add), arg_name = 'filter_add')`.

**Comment by bundfussr** [comment]
> Hi All! I was thinking about this PR last night, and wanted to share two thoughts:
> 
> * If we go the route of using `gsub()`, that should go into the default `arg_name` argument value, rather than into the body of the function. Then if a user ever wants to override the `arg_name` default, the passed value will not be subject to additional parsing.
> * Speaking of the `arg_name` argument: My preference here is now to just do nothing. If we are calling this function AND changing the argument as it's passed to the function (e.g. wrapping the true argument name in `enexpr()`), we can just specify the `arg_name` value. Then this function remains in alignment with every other checking function. For example, `assert_filter_cond(exprs(filter_add), arg_name = 'filter_add')`.

Putting the `gsub()` call into the default value is a good idea. It makes the behavior of the function much clearer.

I would agree to the second point if admiraldev were a general package for argument checking. However, it is intended to support admiral packages. In admiral we have the convention that values don't need to be quoted for arguments which expect a single symbol or expression. I.e., if we do nothing, the default value for `arg_name` in `assert_filter_cond()` and `assert_symbol()` can never be used. Therefore I would prefer setting `arg_name = gsub("^enexpr\\((\\w+)\\)$", "\\1", rlang::caller_arg(arg))` as default value in `assert_filter_cond()`, `assert_symbol()` and `assert_expr()`.

**Comment by ProfessorP-beep** [comment]
Hey @bundfussr I just wanna check before I push. The arg_name changes are only required for those three assert functions? I just saw a similar example in the assertions script, like 'var_name' in 'assert_date_var'.

**Comment by bundfussr** [comment]
> Hey @bundfussr I just wanna check before I push. The arg_name changes are only required for those three assert functions? I just saw a similar example in the assertions script, like 'var_name' in 'assert_date_var'.

I think we don't need to change `assert_date_var()`.

File: R/assertions.R [hunk 2]
```diff
@@ -303,9 +303,14 @@ assert_character_vector <- function(arg, values = NULL, named = FALSE,
 #' If set to `FALSE` and `arg` is `NULL` then an error is thrown. Otherwise,
 #' `NULL` is considered as valid value.
 #' @param arg_name string indicating the label/symbol of the object being checked.
+#'
+#' When the argument for `arg_name` can be input as a quote using
+#' `enexpr()`, i.e. `assert_filter_cond`, `assert_symbol`, and `assert_expr`, then
+#' it is specified without `enexpr()` using `gsub()`.
```
Comment (bundfussr) [inline R/assertions.R:309]: This looks confusing to me. What is the intention? Do we still need it after moving the `gsub()` call to the default value?

Comment (ProfessorP-beep) [reply R/assertions.R:309]: @bundfussr Sorry, was trying to find a way to word it and add the change to the default value so its documented for the user. I can remove it.

Comment (bundfussr) [reply R/assertions.R:309]: I see. It may be helpful for users which are not so familiar with regular expressions.

I would add it only to the three functions where we use `gsub()` in the default value and rephrase the text. What about the following?
> By default the expression specified for `arg` is used. If it is of the form `enexpr(<argument name>)` , the `enexprs()` part is removed. For example if `arg = enexpr(filter_add)` is specified, `arg_name` defaults to `"filter_add"`.

**Review by bundfussr — APPROVED** [review]

**Comment by bms63** [comment]
@ProfessorP-beep are you on pharmaverse slack?  I wanted to ask you a few questions.

File: NEWS.md [hunk 3]
```diff
@@ -4,6 +4,8 @@
 
 ## Updates of Existing Functions
 
+* Updated arg_name parameter in `assert_filter_cond`, `assert_symbol`, and `assert_expr` with  `gsub("^enexpr\\((\\w+)\\)$", "\\1", rlang::caller_arg(arg))` argument so that  error messages return a quoted argument without `enexp()`, e.g. `filter_add` instead of `enexpr(filter_add)`.
```
Comment (bms63) [inline NEWS.md:7]:
```suggestion
* Updated `arg_name` parameter in `assert_filter_cond`, `assert_symbol`, and `assert_expr` with  `gsub("^enexpr\\((\\w+)\\)$", "\\1", rlang::caller_arg(arg))` argument so that  error messages return a quoted argument without `enexpr()`, e.g. `filter_add` instead of `enexpr(filter_add)`.
```

**Review by bms63 — APPROVED** [review]

**Comment by ProfessorP-beep** [comment]
Hey @bms63, sorry since I'm outside the working group list of email domains it won't let me join and says I need a personal invite. I messaged my email on LinkedIn but can post it here as well.
